src/main/java/com/squareup/javapoet/MethodSpec.java

```diff
@@ -68,6 +69,9 @@ private MethodSpec(Builder builder) {
     this.parameters = Util.immutableList(builder.parameters);
     this.varargs = builder.varargs;
     this.exceptions = Util.immutableList(builder.exceptions);
+    this.initializer = (builder.initializer == null)
```

Comment:
please leave this as `null` if there's no initializer. (We should do similarly for field initializers also; I think it's a mistake I made to treat an empty code block as the same as no code block!)